bin/lando/vip-switch.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+set -o errexit -o pipefail -o noclobber -o nounset
+
+# Config vars
+#
+vipswitchpath=/tmp/vip-switch
+wpcontentpath=/app/wp-content
+vipconfigpath=/tmp/vip-switch/vip-config/vip-config.php
+branch=""
+
+# If no arguments are supplied
+#
+if [ $# -eq 0 ]; then
+	read -p "Which site repo do you want to switch to?(e.g: git@github.com:wpcomvip/fake-site.git)? " gitpath
+	read -p "Which branch do you want to use?(defaults to master)" branch
+else
+	PARAMS=""
+	while (( "$#" )); do
+		case "$1" in
+			-b|--branch)
+				if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+					branch=$2
+					shift 2
+				else
+					echo "Error: Argument for $1 is missing" >&2
+					exit 1
+				fi
+				;;
+			-*|--*=) # unsupported flags
+				echo "Error: Unsupported flag $1" >&2
+				exit 1
+				;;
+			*) # preserve positional arguments
+				PARAMS="$PARAMS $1"
+				shift
+				;;
+		esac
+	done	
+
+	gitpath=$PARAMS
+fi
+
+# Default to master branch
+#
+if [ -z $branch ]; then
+	branch=master
+fi
+
+# Trim whitespace
+#
+gitpath=$( echo $gitpath | xargs )
+branch=$( echo $branch | xargs )
+
+# Cleanup if tmp is hanging around for whatever reason 
+#
+rm -rf $vipswitchpath
+
+echo "Cloning repository..."
+git clone --branch $branch --recurse-submodules -j8 $gitpath $vipswitchpath
+
+# Remove possibly conflicting constants
+#
+sed -i.bak -E -e '/USE_VIP_ELASTICSEARCH|VIP_ENABLE_ELASTICSEARCH_QUERY_INTEGRATION|DISALLOW_FILE_EDIT|DISALLOW_FILE_MODS|AUTOMATIC_UPDATER_DISABLED|WP_MAX_MEMORY_LIMIT|VIP_ELASTICSEARCH_ENDPOINTS|VIP_ELASTICSEARCH_USERNAME|VIP_ELASTICSEARCH_PASSWORD|WPCOM_VIP_LOAD_CRON_CONTROL_LOCALLY|WP_CRON_CONTROL_SECRET|WP_DEBUG|WP_DEBUG_LOG/d' $vipconfigpath
```

Comment:
On some repos, this will end up with local changes which may not be ideal. For example, I switched to the wpvip repo and ended up with uncommitted changes, which is unexpected and likely to be accidentally committed.

```
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   vip-config/vip-config.php

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	vip-config/vip-config.php.bak
```

I would go with flagging this as a warning rather than making the change.

Reply:
True. Maybe the solution is to just remove/add it to our wp-config.php depending on if the branch has the constants or not?

Reply:
Is the concern that we end up fatalling/erroring the site due to conflicting constants? Arguably, the constants in vip-config should always have a `defined` check to be on the safe side.

Or some functionality doesn't work correctly? If so, makes sense to add/remove to wp-config (longer-term, could be interesting to have a way to fetch config state via the API).

In any case, I'd suggest that a warning is good enough, but I don't have a strong opinion on this.

Reply:
Oh right. Just need to add defined checks to the generated wp-config.php 🤦 . 1 sec.